src/preprocessing/passes/bool_to_bv.h

```diff
@@ -46,23 +46,72 @@ class BoolToBV : public PreprocessingPass
     ~Statistics();
   };
 
-  /* Takes an assertion and tries to create more bit-vector structure */
-  Node lowerAssertion(const TNode& a);
+  /** Takes an assertion and attempts to create more bit-vector structure
+      by replacing boolean operators with bit-vector operators.
 
-  /* Tries to lower one node to a width-one bit-vector */
-  void lowerNode(const TNode& n);
+      It passes the force argument down to lowerNode, however it never
+      forces the top-level assertion. There's no point forcing the assertion
+      to be a bit-vector when it will just be converted back into a boolean.
+  */
+  Node lowerAssertion(const TNode& node, bool force = false);
+
+  /** Traverses subterms to turn booleans into bit-vectors using lowerNodeHelper
+   *  Passes the force argument to lowerNodeHelper
+   *  Returns the lowered node
+   */
+  Node lowerNode(const TNode& node, bool force = false);
+
+  /** Tries to lower one node to a width-one bit-vector
+   *  Caches the result if successful
+   *
+   *  force = true causes booleans to be converted to bit-vectors using an ITE
+   *     this is only used by mode ALL currently, but could conceivably be used
+   *     in new modes.
```

Comment:
(optional) Perhaps call this parameter `allowIteIntroduction` then (and apply the same name change throughout)?

Reply:
Good idea, done with eeae87d6fb90d88145462825066888f943b6738b